rs/interfaces/src/p2p/consensus.rs

```diff
@@ -137,13 +137,14 @@ pub enum FilterValue {
     Wants,
 }
 
-/// Idempotent and non-blocking function which returns a FilterValue for any artifact ID.
-pub type FilterFn<Id> = Box<dyn Fn(&Id) -> FilterValue + Send + Sync + 'static>;
+/// Idempotent and non-blocking function which returns a BouncerValue for any artifact ID.
+/// Think of this closure as guarding access to the unvalidated pool (similar to a bouncer in a night club).
+pub type Bouncer<Id> = Box<dyn Fn(&Id) -> BouncerValue + Send + Sync + 'static>;
 
-/// Since the FilterFn above is defined as idempotent, the factory trait provides a way to refresh
+/// Since the Bouncer above is defined as idempotent, the factory trait provides a way to refresh
 /// the filter function.
 /// All the filtering logic should happen inside the implentations of the ArtifactAssembler.
-pub trait FilterFnFactory<Artifact: IdentifiableArtifact, Pool>: Send + Sync {
+pub trait BouncerFactory<Artifact: IdentifiableArtifact, Pool>: Send + Sync {
     /// Returns a priority function for the given pool.
```

Comment:
```suggestion
    /// Returns a bouncer function for the given pool.
```

There are quite some more mentions or `priority` and `pfn`. Might be worth to do a `crtl-f` pass over the code base